### ISSUE TITLE
IRIs display error

### ISSUE THREAD

**Issue by mcodescu** [spechub/Hets#1882]
for 
```
logic OWL
ontology DOLCE-Lite =
 <https://ontohub.org/spaceportal/LOA-cnr-it/DOLCE-Lite.owl>
end
```
with Show theory I get a theory that starts with

```logic OWL.NP-SHI

Prefix: : <http://www.loa-cnr.it/ontologies/DOLCE-Lite.owl#>
Prefix: owl: <http://www.w3.org/2002/07/owl#>
Prefix: rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
Prefix: rdfs: <http://www.w3.org/2000/01/rdf-schema#>
Prefix: xsd: <http://www.w3.org/2001/XMLSchema#>

Class: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#particular
Class: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quale
Class: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality-space
Class: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#spatio-temporal-particular
AnnotationProperty: comment
EquivalentClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quale,
                   //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#region
                   and //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#atomic-part-of some //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#region
EquivalentClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality-space,
                   //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#region
                   and //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#overlaps only (not //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality-space)
EquivalentClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#spatio-temporal-particular,
                   //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#particular
                   and (//www.loa-cnr.it/ontologies/DOLCE-Lite.owl#endurant
                        or //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#perdurant
                        or //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality)
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#endurant
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#perdurant
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract-quality,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-quality
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract-quality,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#temporal-quality
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract-region,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-region
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract-region,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#temporal-region
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#amount-of-matter,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#feature
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#amount-of-matter,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-object
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#arbitrary-sum,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#non-physical-endurant
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#arbitrary-sum,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-endurant
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#endurant,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#perdurant
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#endurant,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#feature,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-object
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#non-physical-endurant,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-endurant
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#perdurant,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#quality
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-quality,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#temporal-quality
DisjointClasses: //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#physical-region,
                 //www.loa-cnr.it/ontologies/DOLCE-Lite.owl#temporal-region
```
The class names are displayed without `http:`.

**Comment by mcodescu** [comment]
However, this renaming works
```logic OWL

ontology O1 = 
 Prefix:   : <http://test.org/O1#>
 Prefix:  a: <http://test.org/A#>
 Prefix:  b: <http://test.org/B#>
 
 Class: C
 Class: a:C
 Class: b:C
end

ontology O' =
 O1 with C |-> D
```
but using the full IRI `<http://test.org/O1#C> |-> D` does not:
```Error:
unknown symbol: <http://test.org/O1#C>
in signature:
Prefix: : <<http://test.org/O1#>>
Prefix: a: <<http://test.org/A#>>
Prefix: b: <<http://test.org/B#>>

Class: C
```

**Comment by mcodescu** [comment]
The new branch `iris` has some tracing that is useful.

**Comment by tillmo** [comment]
These are two different issues. Please move the full IRI renaming problem to a different issue. Note that here the Hets parser is involved.

Concerning the first problem that http is dropped: this is not a problem of the Hets parser, because here the OWL API is used (and indeed, if a DOL encapsulating the OWL ontology in Manchester syntax file is imported, the problem disappears). Then the question is: which version of the OWL API do you use?

**Comment by mcodescu** [comment]
You're right, I made a new issue. I use the OWL API that came with Hets: owlapi-osgidistribution-3.5.2.jar. This is quite outdated. It would be good if the Hets package would contain a newer OWL API.

**Comment by tillmo** [comment]
I can reproduce the problem with OWL API version 3.5.2.20150307-2226.
On branch `IRI`, I have added tracing of output of the OWL API. The API version is output at the very end.

**Comment by tillmo** [comment]
The output of OWL API 3.5 looks good. It seems to be a problem of the Hets function that maps the result of the XML parser to an OWL2 theory.

**Comment by mcodescu** [comment]
I have the same OWL API version.

**Comment by tillmo** [comment]
aha, this is `parseProce` in `OWL2/ParseOWL.hs`.

**Comment by tillmo** [comment]
And `parseProc` calls `xmlBasicSpec` in `OWL2/XML.hs`. 
@mcodescu, since this is no longer a parsing but a syntax tree transformation issue, could you please have a look?
The general question is whether we can easily fix the problem, or whether we should switch to OWL API 5.0, with possibly greater changes needed to  `xmlBasicSpec`.

**Comment by mcodescu** [comment]
Yes, I'll see what I can do.

**Comment by tillmo** [comment]
good! #1891 describes an even more fundamental problem

**Comment by mcodescu** [comment]
If I take from DOLCE-Lite only the first class declaration `"#abstract"` I get via `parseIRIReference` the IRI
```
scheme:
no authority
path:
query:
fragment:#abstract
prefix:
isAbbrev:False
```
`appendBase` turns this into

```
scheme:
no authority
path:http://www.loa-cnr.it/ontologies/DOLCE-Lite.owl
query:
fragment:#abstract
prefix:
isAbbrev:False
```

and `splitIRI` puts the http in the prefix
```
scheme:
no authority
path://www.loa-cnr.it/ontologies/DOLCE-Lite.owl
query:
fragment:#abstract
prefix:http
isAbbrev:False
```
According to the comment in `Common/IRI.hs`, `http` should be the scheme not the prefix. Moreover the `://` are lost.

**Comment by mcodescu** [comment]
There's a fix here: https://github.com/spechub/Hets/commit/4035748faaca2cb25ce1e1b284bd5af0c03fdb33

A meta-question: do we want to display the expanded IRIs `http://www.loa-cnr.it/ontologies/DOLCE-Lite.owl#abstract` or just `abstract` would do? The empty prefix is available.

**Comment by tillmo** [comment]
could you turn your fix into a pull request, please?

**Comment by mcodescu** [comment]
Now I see there is a pull request: https://github.com/spechub/Hets/pull/1909

**Comment by mcodescu** [comment]
Solved on #1909
